java/src/main/java/com/saucelabs/simplesauce/SauceOptions.java

```diff
@@ -0,0 +1,148 @@
+package com.saucelabs.simplesauce;
+
+import lombok.Getter;
+import lombok.Setter;
+import org.openqa.selenium.chrome.ChromeOptions;
+import org.openqa.selenium.edge.EdgeOptions;
+import org.openqa.selenium.ie.InternetExplorerOptions;
+
+public class SauceOptions {
+    //TODO can probably use BrowserType enum from Selenium to do BrowserType.CHROME
+    @Getter
+    @Setter
+    public String browser = "Chrome";
+
+    @Getter @Setter public String browserVersion = "latest";
+    @Getter @Setter public String operatingSystem = "Windows 10";
+    @Getter public  ChromeOptions chromeOptions;
+    private EdgeOptions edgeOptions;
+    private InternetExplorerOptions ieOptions;
+
+    public SauceOptions withChrome()
+    {
+        chromeOptions = new ChromeOptions();
+        chromeOptions.setExperimentalOption("w3c", true);
+        browser = "Chrome";
+        return this;
+    }
+    public SauceOptions withSafari()
+    {
+        return withMacOsMojave();
+    }
+    public SauceOptions withSafari(String version)
+    {
+        //TODO: I did this but I hate it :(
+        //I wish I could just have a default value set for the version param
+        if(version.equalsIgnoreCase(""))
+            version = "latest";
+        browser = "safari";
+        browserVersion = version;
+        return this;
+    }
+    public SauceOptions withMacOsMojave() {
+        operatingSystem = "macOS 10.14";
+        browser = "safari";
+        browserVersion = "12.0";
+        return this;
+    }
+    public SauceOptions withMacOsHighSierra()
+    {
+        operatingSystem = "macOS 10.13";
+        browser = "Safari";
+        return this;
+    }
+    public SauceOptions withMacOsSierra() {
+        operatingSystem = "macOS 10.12";
+        browser = "Safari";
+        return this;
+    }
+
+    public SauceOptions withMacOsXElCapitan() {
+        operatingSystem = "OS X 10.11";
+        browser = "Safari";
+        return this;
+    }
+
+    public SauceOptions withMacOsXYosemite() {
+        operatingSystem = "OS X 10.10";
+        browser = "Safari";
+        return this;
+    }
+    public SauceOptions withLinux() {
+        operatingSystem = "Linux";
+        return this;
+    }
+
+    public SauceOptions withWindows10() {
+        operatingSystem = "Windows 10";
+        return this;
+    }
+    public SauceOptions withWindows8_1() {
+        operatingSystem = "Windows 8.1";
+        return this;
+    }
+    public SauceOptions withWindows8() {
+        operatingSystem = "Windows 8";
+        return this;
+    }
+
+    public SauceOptions withWindows7() {
+        operatingSystem = "Windows 7";
+        return this;
+    }
+
+    public SauceOptions withEdge() {
```

Comment:
If this method is public maybe we should add the latest browserVersion too and change it if specified? Or have it private?

Reply:
@TheTestLynx can you elaborate a bit more? Are you saying to be able to pass in a `edgeVersion` into this method?